src/object.c

```diff
@@ -631,6 +631,10 @@ void trimStringObjectIfNeeded(robj *o, int trim_small_values) {
     }
 }
 
+int canUseSharedObject(void) {
+    return server.maxmemory == 0 || !(server.maxmemory_policy & MAXMEMORY_FLAG_NO_SHARED_INTEGERS);
+}
```

Comment:
I think we can make this `static inline` and put it in server.h

Reply:
A test case is added.